### DISCUSSION TITLE
Easier way to do a bunch of incompatibilities

### DISCUSSION THREAD

**Discussion by jewvenchy** [nftchef/art-engine#192] [Help + Q&A]
i was wondering if there's a way to set incompatibilities but for a whole batch

for example I have sets of color divided into 3 "types" ie base, bright, soft.

inside of these color sets there are 5 diff traits exported in like 20+ colors specific to that color set. so it's:

```
base > trait A > 20 colors
base > trait B > 20 colors
bright > trait A > 20 colors
bright > trait B > 20 colors
soft > trait A > 20 colors
soft > trait B > 20 colors
and so on for 5 traits (A through E)
```

i want it so that traits A-E are never the same color. for example:

token 1 is the base color set, it has trait A as color 1, so trait B-E can never be color 1
token 1 trait B is color 5, so trait C-E can't be color 1 or color 5

is there a better way to do this or anything I can do besides doing it individually for literally every single layer?

ie:

```js
const incompatible = {
Trait A Red: ["Trait B Red", "Trait C Red", "Trait D Red", "Trait E Red"],
"Trait B Red": ["Trait A Red", "Trait C Red", "Trait D Red", "Trait E Red"],
"Trait C Red": ["Trait A Red", "Trait B Red", "Trait D Red", "Trait E Red"],
};
``` 

thank you very much

**Comment by BakeSloth** [comment]
I'm not sure, but I would try separating each trait by color into their own folders. keeping the name of the color folders the same.

-Trait A
--Red
--Blue
--Green
-Trait B
--Red
--Blue
--Green
-Trait C
--Red
--Blue
--Green

const incompatible = {
Red: ["Red"],
Blue: ["Blue"],
Green: ["Green"],
};

This way, whatever trait color comes first, won't come up again in higher traits.

let us know if it works!

**Reply by cryptofannatic** [reply]
@BakeSloth you mean like this? 
![image](https://user-images.githubusercontent.com/109002076/178151294-2fdd6b2e-779c-49e2-a525-addbddd76e99.png)
Does the sub folder for blue colour need a weight too?
